source/git/gitsource_unix.go

```diff
@@ -4,13 +4,41 @@ package git
 
 import (
 	"context"
+	"fmt"
+	"os"
 	"os/exec"
 	"syscall"
 	"time"
+
+	"github.com/docker/docker/pkg/reexec"
+	"golang.org/x/sys/unix"
 )
 
+func init() {
+	reexec.Register("reexec-git", reexecGit)
+}
+
+func reexecGit() {
+	// Need standard user umask for git process.
+	unix.Umask(0022)
+
+	// Reexec git command
+	cmd := exec.Command("git", os.Args[2:]...)
+	cmd.Stdout = os.Stdout
+	if err := cmd.Run(); err != nil {
```

Comment:
does this need signal forwarding?

Reply:
Yes we need to forward signals to the subproc